src/Resources/config/core.php

```diff
@@ -0,0 +1,232 @@
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Sonata\AdminBundle\Admin\AdminHelper;
+use Sonata\AdminBundle\Admin\BreadcrumbsBuilder;
+use Sonata\AdminBundle\Admin\BreadcrumbsBuilderInterface;
+use Sonata\AdminBundle\Admin\Extension\LockExtension;
+use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Controller\HelperController;
+use Sonata\AdminBundle\Event\AdminEventExtension;
+use Sonata\AdminBundle\Export\Exporter;
+use Sonata\AdminBundle\Filter\FilterFactory;
+use Sonata\AdminBundle\Filter\FilterFactoryInterface;
+use Sonata\AdminBundle\Filter\Persister\FilterPersisterInterface;
+use Sonata\AdminBundle\Filter\Persister\SessionFilterPersister;
+use Sonata\AdminBundle\Model\AuditManager;
+use Sonata\AdminBundle\Model\AuditManagerInterface;
+use Sonata\AdminBundle\Route\AdminPoolLoader;
+use Sonata\AdminBundle\Search\SearchHandler;
+use Sonata\AdminBundle\Templating\MutableTemplateRegistryInterface;
+use Sonata\AdminBundle\Templating\TemplateRegistry;
+use Sonata\AdminBundle\Translator\BCLabelTranslatorStrategy;
+use Sonata\AdminBundle\Translator\Extractor\AdminExtractor;
+use Sonata\AdminBundle\Translator\Extractor\JMSTranslatorBundle\AdminExtractor as DeprecatedAdminExtractor;
+use Sonata\AdminBundle\Translator\FormLabelTranslatorStrategy;
+use Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface;
+use Sonata\AdminBundle\Translator\NativeLabelTranslatorStrategy;
+use Sonata\AdminBundle\Translator\NoopLabelTranslatorStrategy;
+use Sonata\AdminBundle\Translator\UnderscoreLabelTranslatorStrategy;
+use Sonata\AdminBundle\Twig\GlobalVariables;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ReferenceConfigurator;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    // Use "service" function for creating references to services when dropping support for Symfony 4.4
+    // Use "param" function for creating references to parameters when dropping support for Symfony 5.1
+    $containerConfigurator->services()
+
+        ->set('sonata.admin.pool', Pool::class)
+            ->public()
+            ->args([
+                new ReferenceConfigurator('service_container'),
+                '',
+                '',
+                [],
+                new ReferenceConfigurator('property_accessor'),
+            ])
+            ->call('setTemplateRegistry', [
+                new ReferenceConfigurator('sonata.admin.global_template_registry'),
+            ])
+
+        ->alias(Pool::class, 'sonata.admin.pool')
+
+        ->set('sonata.admin.route_loader', AdminPoolLoader::class)
+            ->public()
+            ->tag('routing.loader')
+            ->args([
+                new ReferenceConfigurator('sonata.admin.pool'),
+                '',
```

Comment:
this is strange, This constructor accept only array, maybe using null is better for empty arguments?

Reply:
I've changed it to array as the constructor expects, I guess it doesn't really matter because they will be replaced in a compiler pass.